### PR TITLE
[VecOps] Add Argsort with custom comparator

### PR DESCRIPTION
**Pull request by stwunsch**
Feature + tests!

### CODE REVIEW

File: math/vecops/inc/ROOT/RVec.hxx [hunk 1]
```diff
@@ -1009,6 +1011,29 @@ RVec<typename RVec<T>::size_type> Argsort(const RVec<T> &v)
    return i;
 }
 
+/// Return an RVec of indices that sort the input RVec based on a comparison operator
```
Comment (eguiraud) [inline math/vecops/inc/ROOT/RVec.hxx:1014]:
```suggestion
/// Return an RVec of indices that sort the input RVec based on a comparison function.
```

File: math/vecops/inc/ROOT/RVec.hxx [hunk 2]
```diff
@@ -1009,6 +1011,29 @@ RVec<typename RVec<T>::size_type> Argsort(const RVec<T> &v)
...
+///
+/// Example code, at the ROOT prompt:
+/// ~~~{.cpp}
+/// using namespace ROOT::VecOps;
+/// RVec<double> v {2., 3., 1.};
+/// auto sortIndices = Argsort(v, [](double x, double y) {return x > y;});
+/// sortIndices
+/// // (ROOT::VecOps::RVec<unsigned long> &) { 1, 0, 2 }
+/// Take(v, sortIndices);
+/// // (ROOT::VecOps::RVec<double> &) { 3., 2., 1. }
```
Comment (eguiraud) [inline math/vecops/inc/ROOT/RVec.hxx:1024]: Simplified a bit:

```suggestion
/// auto sortIndices = Argsort(v, [](double x, double y) {return x > y;})
/// (ROOT::VecOps::RVec<unsigned long> &) { 1, 0, 2 }
/// Take(v, sortIndices)
/// (ROOT::VecOps::RVec<double> &) { 3., 2., 1. }
```

File: math/vecops/inc/ROOT/RVec.hxx [hunk 3]
```diff
@@ -998,6 +998,8 @@ void swap(RVec<T> &lhs, RVec<T> &rhs)
 /// auto sortIndices = Argsort(v);
 /// sortIndices
 /// // (ROOT::VecOps::RVec<unsigned long> &) { 2, 0, 1 }
+/// Take(v, sortIndices)
+/// // (ROOT::VecOps::RVec<double> &) { 1., 2., 3. }
```
Comment (eguiraud) [inline math/vecops/inc/ROOT/RVec.hxx:1002]:
```suggestion
/// (ROOT::VecOps::RVec<double> &) { 1., 2., 3. }
```
why the extra `//`?

**Review by eguiraud — APPROVED** [review]
Thanks @stwunsch , just minor comments on the docstrings

File: math/vecops/inc/ROOT/RVec.hxx (hunk 3 above)
Comment (stwunsch) [reply math/vecops/inc/ROOT/RVec.hxx:1002]: It's then a comment in the code block and not actual code!

File: math/vecops/inc/ROOT/RVec.hxx (hunk 2 above)
Comment (stwunsch) [reply math/vecops/inc/ROOT/RVec.hxx:1024]: Good idea! Do we want the printout of the type and data (like in the ROOT prompt) as code or commented with the `//` in front?

File: math/vecops/inc/ROOT/RVec.hxx (hunk 3 above)
Comment (stwunsch) [reply math/vecops/inc/ROOT/RVec.hxx:1002]: You can have a look here how it looks: https://root.cern/doc/master/namespaceROOT_1_1VecOps.html#a7dcd060b97f6c82621ba0d8f376ad195

Comment (eguiraud) [reply math/vecops/inc/ROOT/RVec.hxx:1002]: ah I see that's the style used for everything except https://root.cern/doc/master/namespaceROOT_1_1VecOps.html#abbbf82257156b750c7b9bdd39a53ee33 , ok then

File: math/vecops/inc/ROOT/RVec.hxx (hunk 2 above)
Comment (eguiraud) [reply math/vecops/inc/ROOT/RVec.hxx:1024]: let's keep the `//` for consistency with the rest

**Pull request merged**